libraries/unit_converter.rb

```diff
@@ -0,0 +1,40 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# File Name:: unit_converter.rb
+#
+# Module contains different functions used to manipulate the units of file sizes.
+#
+module CernerSplunk
+  SIZE_SCALE = %w(KB MB GB TB).freeze
+  # Methods converts file sizes in KB, MB, GB and TB into Bytes.
+  def self.convert_to_bytes(string)
+    fail "Unparsable size input: #{string}" if string.match(/^\d+\w{2,3}$/).nil?
+
+    size = string.scan(/\d+|\w/)[0]
+    size_unit = string.scan(/\d+|\w/)[1].upcase
+
+    case size_unit
+    when 'K'
+      size.to_i * 1024
+    when 'M'
+      size.to_i * 1024**2
+    when 'G'
+      size.to_i * 1024**3
+    when 'T'
+      size.to_i * 1024**4
+    else
+      fail "#{string} :Please specify the size in KB, MB, GB or TB"
+    end
+  end
+
+  # Function returns the size in a human readable format.
+  def self.human_readable_size(filesize, level = 0)
```

Comment:
Minor but instead of worrying about recursion, couldn't this be done in a while loop that simply increments a local variable `level` that starts out as zero using your same conditional?